### PR TITLE
INT B-21393 Consolidate PPM Allowable Weights

### PR DESCRIPTION
**Pull request by ajlusk**
## Summary

Moves the allowable weight column from weight_tickets to ppm_shipments to that there is one consolidated allowable weight per shipment rather than one per trip.

The initial allowable weight is now calculated when the customer submits their PPM for closeout rather than in the UI as the SC opens each weight ticket.

The new allowable weight is editable by the SC when reviewing weight tickets.

This is the first time I've done a non-trivial database migration so pay extra attention to that in your review. The backfill may not be necessary. I added it to keep existing test data usable.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the customer app, create a new move with at least one PPM, submit it.
2. Login as a Services Counselor and approve the move.
3. Log back in as your original customer and upload weight tickets for at least two trips and submit the shipment for closeout. Take note of the new weight of both trips.
4. Log back in as a Services Counselor and review the documents for the move you just submitted for closeout.
5. Verify the "Shipment Info" header section is expanded upon opening the document viewer.
6. Verify the above section contains a field labeled "Allowable Weight"
7. Verify the weight ticket section below no longer contains an "Allowable Weight" section.
8. Hover over the info icon next to the allowable weight, verify a tooltip with text matching the AC appears.
9. Verify you can edit the Allowable weight in the header section.
10. Run ANDI, the new button should be focusable and have an accessible name/label
11. Approve the submitted documents.
12. Navigate to the Move history, verify the change you made to the allowable weight value is listed.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2024-10-10 at 10 11 27 AM
Screenshot 2024-10-10 at 10 11 41 AM
Screenshot 2024-10-10 at 10 12 29 AM

### CODE REVIEW

File: src/components/Office/PPM/PPMHeaderSummary/EditPPMHeaderSummaryModal.jsx [hunk 1]
```diff
@@ -54,6 +56,14 @@ const EditPPMHeaderSummaryModal = ({ sectionType, sectionInfo, onClose, onSubmit
     }),
   });
 
+  const weightValidationSchema = Yup.object().shape({
```
Comment (MInthavongsay) [inline src/components/Office/PPM/PPMHeaderSummary/EditPPMHeaderSummaryModal.jsx:59]: You need to add a validation to check for negative numbers or 0. Tried clicking <Save> with bad number and it doesn't do anything which is minimally good. An alert saying negative numbers are not allowed would be more user friendly here.
Screenshot 2024-10-10 at 11 59 33 AM

**Comment by CoryKleinjanCACI** [comment]
The allowable weight section after approving the shipment weights is blank for me.

**Comment by CoryKleinjanCACI** [comment]
The allowable weight section after approving the shipment weights is blank for me.
Screenshot 2024-10-10 at 12 20 32 PM

**Comment by ajlusk** [comment]
> The allowable weight section after approving the shipment weights is blank for me. Screenshot 2024-10-10 at 12 20 32 PM

Resolved.

That should have been removed. Lost that change in a stash while trying to deal with not being able to commit after the go version change.

File: src/components/Office/PPM/PPMHeaderSummary/EditPPMHeaderSummaryModal.jsx (hunk 1 above)
Comment (ajlusk) [reply src/components/Office/PPM/PPMHeaderSummary/EditPPMHeaderSummaryModal.jsx:59]: This validation has been added.

**Review by CoryKleinjanCACI — APPROVED** [review]
Looks good after the added validation and cleanup.

**Review by MInthavongsay — APPROVED** [review]
Looks good!

**Review by deandreJones — APPROVED** [review]

**Review by begrohmann — APPROVED** [review]

**Comment by r-mettler** [comment]
Looks good

**Review by r-mettler — APPROVED** [review]

**Pull request merged**